unity-renderer/Assets/Scripts/MainScripts/DCL/Controllers/BuilderMode/Analytics/BIWAnalytics.cs

```diff
@@ -0,0 +1,201 @@
+
+    /// <summary>
+    /// Everytime we start the flow of the editor
+    /// </summary>
+    /// <param name="source">It comes from the shortcut or from BuilderPanel</param>
+    public static void StartEditorFlow(string source)
+    {
+        Dictionary<string, string> events = new Dictionary<string, string>();
+        events.Add("source", source);
+        AddLandInfoSendEvent("start_editor_flow", events);
+    }
+
+    public static void EnterEditor(float loadingTime)
+    {
+        Dictionary<string, string> events = new Dictionary<string, string>();
+        events.Add("loading_time", loadingTime.ToString());
+        AddLandInfoSendEvent("enter_editor", events);
+    }
+
+    public static void ExitEditor(float timeInvestedInTheEditor)
+    {
+        Dictionary<string, string> events = new Dictionary<string, string>();
+        events.Add("time_in_the_editor", timeInvestedInTheEditor.ToString());
+        AddLandInfoSendEvent("exit_editor", events);
+    }
+
+    public static void StartScenePublish(SceneMetricsModel sceneLimits)
+    {
+        Dictionary<string, string> events = new Dictionary<string, string>();
+        events.Add("scene_limits", ConvertSceneMetricsModelToDictionary(sceneLimits).ToString());
+        AddLandInfoSendEvent("start_publish_of_the_scene", events);
+    }
+
+    public static void EndScenePublish(SceneMetricsModel sceneLimits, string successOrError, float publicationTime)
+    {
+        Dictionary<string, string> events = new Dictionary<string, string>();
+        events.Add("success", successOrError);
+        events.Add("publication_time", publicationTime.ToString());
+        events.Add("scene_limits", ConvertSceneMetricsModelToDictionary(sceneLimits).ToString());
+        AddLandInfoSendEvent("end_scene_publish", events);
+    }
+
+    public static void SceneLimitsOverPassed(SceneMetricsModel sceneLimits)
+    {
+        Dictionary<string, string> events = new Dictionary<string, string>();
+        events.Add("scene_limits", ConvertSceneMetricsModelToDictionary(sceneLimits).ToString());
+        AddLandInfoSendEvent("scene_limits_over_passed", events);
+    }
+
+    /// <summary>
+    /// When a new item is placed from the catalog
+    /// </summary>
+    /// <param name="catalogItem">The item that has been added</param>
+    /// <param name="source">It has been added from Categories, Asset pack, Favorites or Quick Access</param>
+    public static void NewObjectPlaced(CatalogItem catalogItem, string source)
+    {
+        Dictionary<string, string> events = new Dictionary<string, string>();
+        events.Add("name", catalogItem.name);
+        events.Add("assetPack", catalogItem.assetPackName);
+        events.Add("category", catalogItem.category);
+        events.Add("category Name", catalogItem.categoryName);
+        events.Add("source", source);
+        events.Add("type", catalogItem.itemType.ToString());
+        AddLandInfoSendEvent("new_object_placed", events);
```

Comment:
Could this be a little bit heavy if the user is putting e.g. a wall around? Maybe we should only send this if it's different to the last object placed? So we don't get 100x the same object because they are planting flowers or building walls in a scene?

Reply:
If the user is putting severals wall this will be fired each time.  I think that most of the user won't set more than 100 assets per scene, but this is only a "non-exact prediction".

 If we only send 1 event for each asset we lose the ability to track the most used asset in total. Also, we are losing how many times a user is using an asset per scene but this may be overkill. 

Another option is to add a counter with the number of times that the user has set the asset and only send the event if X seconds has pass without set the same asset in the scene, but we will lose some information if the user disable biw or exit from the explorer

@olavra what do you think about that?

Reply:
>  we are losing how many times a user is using an asset per scene 
We'll still have this data in the finished scene right? I plan on downloading all scenes anyway to the DWH, so you don't really need this data sent by Segment. I was just thinking if you can save in a variable the last object sent, and if last_asset = current_asset, then don't send, otherwise send. That way if someone does:

Place 100 walls
Place 50 flowers
Place 25 walls

We'd only get 3 events (place wall, place flower, place wall) - Makes sense? I don't think we need 175 events for this scenario above

Reply:
Yep but when do you send that event to be sure that we are only sending 3 events? Users can close the web any time without exiting from the editor and the others parts of biw will work.

We can send the event if nothing has changed in X seconds. Does that work for you? Which time will you propose?

Reply:
I like the idea of sending batches every few seconds.

Another alternative is reporting when you change your selected asset or close the BiW. There's the downside of losing track of the lasts placed items if the user were to close the tab, but I think it's a fair trade-off.